client/my-sites/site-settings/date-time-format/date-time-format-options.jsx

```diff
@@ -0,0 +1,338 @@
+const defaultTimeFormats = [ 'g:i a', 'g:i A', 'H:i' ];
+
+export class DateTimeFormatOptions extends Component {
+	state = {
+		customDateFormat: false,
+		customTimeFormat: false,
+		isLoadingSettings: true,
+	};
+
+	isDefaultFormat = ( defaultFormats, format ) => ! includes( defaultFormats, format );
+
+	handleRadio = event => {
+		const { name: field, value: format } = event.currentTarget;
+		this.props.handleRadio( event );
+		if ( 'date_format' === field ) {
+			this.setState( {
+				customDateFormat: this.isDefaultFormat( defaultDateFormats, format ),
+			} );
+		} else if ( 'time_format' === field ) {
+			this.setState( {
+				customTimeFormat: this.isDefaultFormat( defaultTimeFormats, format ),
+			} );
+		}
+	}
+
+	handleInput = field => event => {
+		this.props.onChangeField( field )( event );
+		if ( 'date_format' === field ) {
+			this.setState( { customDateFormat: true } );
+		} else if ( 'time_format' === field ) {
+			this.setState( { customTimeFormat: true } );
+		}
+	}
+
+	handleCustomRadio = event => {
+		const { name: field } = event.currentTarget;
+		if ( 'date_format' === field ) {
+			this.setState( { customDateFormat: true } );
+		} else if ( 'time_format' === field ) {
+			this.setState( { customTimeFormat: true } );
+		}
+	}
+
+	componentWillReceiveProps( newProps ) {
+		const {
+			fields: {
+				date_format: dateFormat,
+				time_format: timeFormat,
+			},
+		} = newProps;
+
+		if ( ! this.state.isLoadingSettings || '' === dateFormat || '' === timeFormat ) {
+			return;
+		}
+
+		this.setState( {
+			customDateFormat: this.isDefaultFormat( defaultDateFormats, dateFormat ),
+			customTimeFormat: this.isDefaultFormat( defaultTimeFormats, timeFormat ),
+			isLoadingSettings: false,
+		} );
+	}
+
+	dateFormatOption() {
+		const {
+			fields: {
+				date_format: dateFormat,
+				timezone_string: timezoneString,
+			},
+			isRequestingSettings,
+			moment,
+			translate,
+		} = this.props;
+		const { customDateFormat: isCustomFormat } = this.state;
+
+		const today = startsWith( timezoneString, 'UTC' )
+			? moment().utcOffset( timezoneString.substring( 3 ) * 60 )
+			: moment.tz( timezoneString );
+
+		return (
+			<FormFieldset>
+				<FormLabel>
+					{ translate( 'Date Format' ) }
+				</FormLabel>
+				{ defaultDateFormats.map( ( format, key ) =>
+					<FormLabel key={ key }>
+						<FormRadio
+							checked={ ! isCustomFormat && format === dateFormat }
+							disabled={ isRequestingSettings }
+							name="date_format"
+							onChange={ this.handleRadio }
+							value={ format }
+						/>
+						<span>{ today.format( phpToMomentDatetimeFormat( format ) ) }</span>
+					</FormLabel>
+				) }
+				<FormLabel className="date-time-format__custom-field">
+					<FormRadio
+						checked={ isCustomFormat }
+						disabled={ isRequestingSettings }
+						name="date_format"
+						onChange={ this.handleCustomRadio }
+						value={ dateFormat }
+					/>
+					<span>
+						{ translate( 'Custom' ) }
```

Comment:
this could certainly use a comment and maybe even a `context` since it's not clear which meaning of `Custom` we mean.

```js
// maybe?
translate( 'Custom', { comment: 'Indicates a custom value' } )
```